make/autoconf/flags-cflags.m4

```diff
@@ -881,3 +883,30 @@ AC_DEFUN([FLAGS_SETUP_GCC6_COMPILER_FLAGS],
       PREFIX: $2, IF_FALSE: [NO_LIFETIME_DSE_CFLAG=""])
   $1_GCC6_CFLAGS="${NO_DELETE_NULL_POINTER_CHECKS_CFLAG} ${NO_LIFETIME_DSE_CFLAG}"
 ])
+
+AC_DEFUN_ONCE([FLAGS_SETUP_BRANCH_PROTECTION],
+[
+  # Is branch protection available?
+  BRANCH_PROTECTION_AVAILABLE=true
+  BRANCH_PROTECTION_FLAG="-mbranch-protection=standard"
+
+  AC_MSG_CHECKING([if branch protection is available])
+  if test "x$OPENJDK_TARGET_CPU" != xaarch64; then
+    AC_MSG_RESULT([no, does not work for $OPENJDK_TARGET_CPU])
+    BRANCH_PROTECTION_AVAILABLE=false
+  elif test "x$TOOLCHAIN_TYPE" = xgcc || test "x$TOOLCHAIN_TYPE" = xclang; then
+    # Check that the compiler actually supports branch protection.
+    FLAGS_COMPILER_CHECK_ARGUMENTS(ARGUMENT: [${BRANCH_PROTECTION_FLAG}],
```

Comment:
This branch misses a AC_MSG_RESULT, which prints the newline. The resulting output will look messy.

Reply:
Looking at this block of code again, I've got far too many outputted lines compared to other features. Removing some means I can simplify the code too, so I'll do that.